Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -101,6 +106,109 @@ public virtual Fill Fill(FillModelParameters parameters)
+        ///
+        ///     With limit if touched, we can't be sure of the order of the H - L values for the limit fill. The assumption
+        ///     was made the limit fill will be done with closing price of the bar after the stop has been triggered.
+        /// </remarks>
+        public virtual OrderEvent LimitIfTouchedFill(Security asset, LimitIfTouchedOrder order)
+        {
+            //Default order event to return.
+            var utcTime = asset.LocalTime.ConvertToUtc(asset.Exchange.TimeZone);
+            var fill = new OrderEvent(order, utcTime, OrderFee.Zero);
+
+            //If its cancelled don't need anymore checks:
+            if (order.Status == OrderStatus.Canceled) return fill;
+
+            // make sure the exchange is open/normal market hours before filling
+            if (!IsExchangeOpen(asset, false)) return fill;
+
+            // Get the range of prices in the last bar:
+            var high = 0m;
+            var low = 0m;
+            var open = 0m;
+            var endTimeUtc = DateTime.MinValue;
+
+            var subscribedTypes = GetSubscribedTypes(asset);
+            var slip = asset.SlippageModel.GetSlippageApproximation(asset, order); 
+            
+            if (subscribedTypes.Contains(typeof(Tick)))
+            {
+                var trade = asset.Cache.GetAll<Tick>().LastOrDefault(x => x.TickType == TickType.Trade && x.Price > 0);
+                if (trade != null)
+                {
+                    high = trade.Price;
+                    low = trade.Price;
+                    open = trade.Price;
+                    endTimeUtc = trade.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+            else if (subscribedTypes.Contains(typeof(TradeBar)))
+            {
+                var tradeBar = asset.Cache.GetData<TradeBar>();
+                if (tradeBar != null)
+                {
+                    high = tradeBar.High;
+                    low = tradeBar.Low;
+                    open = tradeBar.Open;
+                    endTimeUtc = tradeBar.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+                }
+            }
+            //Get the range of prices in the last bar:
+
+            // do not fill on stale data
+            if (endTimeUtc <= order.Time) return fill;
+
+            //Check if the limit if touched order was filled:
+            switch (order.Direction)
+            {
+                case OrderDirection.Buy:
+                    //-> 1.2 Buy: If Price below Trigger, Buy:
+                    if (low <= order.TriggerPrice || order.TriggerTouched)
+                    {
+                        order.TriggerTouched = true;
+
+                        if (low <= order.LimitPrice)
+                        {
+                            fill.Status = OrderStatus.Filled;
+                            fill.FillPrice = open >= order.LimitPrice? open + slip // Fill price conditional on open
```

Comment:
Here we are buying: `open >= order.LimitPrice? open + slip // Fill price conditional on open` the limit price can't fill worse than the limit price, it can fill better though: `Using a Limit if Touched order helps to ensure that, if the order does execute, the order will not execute at a price less favorable than the limit price.` at https://www1.interactivebrokers.com/en/index.php?f=592
Behavior should be covered by unit tests 👍

Reply:
Good eye, should have been a `<=` !